pkg/tools/cli/loadtest/service/loader.go

```diff
@@ -138,44 +165,108 @@ func (l *loader) Do(ctx context.Context) <-chan error {
+					return new(payload.Search_Response)
+				}
+			}
+
+			start = time.Now()
+			eg.Go(safety.RecoverFunc(func() error {
+				// TODO: related to #557
+				/*
 					_, err := l.client.Do(egctx, l.addr, func(ctx context.Context, conn *grpc.ClientConn, copts ...grpc.CallOption) (interface{}, error) {
-						err := l.loaderFunc(ctx, vald.NewValdClient(conn), r, copts...)
-						atomic.AddInt32(&pgCnt, 1)
+						st, err := l.loaderFunc(ctx, conn, nil, copts...)
 						if err != nil {
-							log.Warn("an error occurred while executing loaderfunc:", err)
-							atomic.AddInt32(&errCnt, 1)
+							return nil, err
 						}
-						return nil, err
+						return nil, grpc.BidirectionalStreamClient(st.(grpc.ClientStream), l.dataProvider, newData, f)
 					})
+				*/
+				conn, err := grpc.Dial(l.addr, grpc.WithInsecure())
+				if err != nil {
+					finalize(err)
+					return nil
+				}
+				defer func() {
+					finalize(conn.Close())
+				}()
+				st, err := l.loaderFunc(egctx, conn, nil)
+				if err != nil {
+					finalize(err)
+					return nil
+				}
+				if err := igrpc.BidirectionalStreamClient(st.(grpc.ClientStream), l.dataProvider, newData, f); err != nil {
+					finalize(err)
+				}
+				return nil
+			}))
+			err = eg.Wait()
+			end = time.Now()
+		case config.Insert, config.Search:
+			eg.Limitation(l.concurrency)
+			start = time.Now()
+			for {
+				r := l.dataProvider()
+				if r == nil {
+					break
+				}
+				eg.Go(safety.RecoverFunc(func() error {
+					// TODO: related to #557
+					/*
+						_, err := l.client.Do(egctx, l.addr, func(ctx context.Context, conn *grpc.ClientConn, copts ...grpc.CallOption) (interface{}, error) {
+							res, err := l.loaderFunc(ctx, conn, r, copts...)
+							f(res, err)
+							return res, err
+						})
+					*/
+					conn, err := grpc.Dial(l.addr, grpc.WithInsecure())
 					if err != nil {
-						select {
-						case <-ctx.Done():
-							ech <- errors.Wrap(err, ctx.Err().Error())
-						case ech <- err:
-						}
+						finalize(err)
+						return nil
 					}
+					defer func() {
+						finalize(conn.Close())
+					}()
+					res, err := l.loaderFunc(ctx, conn, r)
+					f(res, err)
+					if err != nil {
+						finalize(err)
+					}
+
 					return nil
 				}))
-		}
-		if err := eg.Wait(); err != nil {
-			select {
-			case <-ctx.Done():
-				ech <- errors.Wrap(err, ctx.Err().Error())
-			case ech <- err:
 			}
+			err = eg.Wait()
+			end = time.Now()
+		default:
+			err = errors.Errorf("undefined type: %s", l.operation.String())
+		}
+		ticker.Stop()
+
+		if errCnt > 0 {
+			log.Warnf("Error ratio: %.2f%%", float64(errCnt)/float64(pgCnt)*100)
+			err = errors.Errorf("insert failure: %d", errCnt)
+		}
+		if err != nil {
+			finalize(err)
 			return p.Signal(syscall.SIGKILL) // TODO: #403
 		}
-		log.Infof("Error ratio: %.2f%%", float64(errCnt)/float64(pgCnt)*100)
+		log.Infof("result:%s\t%d\t%d\t%f", l.service.String(), l.concurrency, l.batchSize, vps(int(pgCnt)*l.batchSize, start, end))
```

Comment:
```suggestion
		log.Infof("result: Service: %s, Concurrency: %d, Batch Size: %d, VPS: %f", l.service.String(), l.concurrency, l.batchSize, vps(int(pgCnt)*l.batchSize, start, end))
```

Reply:
Thank you for your comment.
I use this log for drawing charts, so tab separated is useful for that.